src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -405,6 +405,15 @@ private static void AddActionDataToBuilder(
                 builder.Metadata.Add(filter);
             }
         }
+        action.EndpointMetadata ??= [];
+        foreach (var metadata in builder.Metadata.OfType<IProducesResponseTypeMetadata>())
+        {
+            if (action.EndpointMetadata
```

Comment:
This is Linq, to do not override the types that are passed though attributes (To do not break any existing client).
I know that you are avoiding Linq for performance reasons.